cmd/labs/project/project.go

```diff
@@ -0,0 +1,294 @@
+	authFile := p.authFile(ctx)
+	log.Debugf(ctx, "Loading auth configuration from: %s", authFile)
+	return tryLoadAndParseJSON[authConfig](authFile)
+}
+
+func (p *Project) writeAuthConfig(ctx context.Context, ac authConfig) error {
+	authFile := p.authFile(ctx)
+	raw, err := json.Marshal(ac)
+	if err != nil {
+		return err
+	}
+	log.Debugf(ctx, "Writing auth configuration to: %s", authFile)
+	return os.WriteFile(authFile, raw, 0o600)
+}
+
+func (p *Project) versionFile(ctx context.Context) string {
+	return filepath.Join(p.StateDir(ctx), "version.json")
+}
+
+func (p *Project) InstalledVersion(ctx context.Context) (*version, error) {
+	versionFile := p.versionFile(ctx)
+	log.Debugf(ctx, "Loading installed version info from: %s", versionFile)
+	return tryLoadAndParseJSON[version](versionFile)
+}
+
+func (p *Project) writeVersionFile(ctx context.Context, ver string) error {
+	versionFile := p.versionFile(ctx)
+	raw, err := json.Marshal(version{
+		Version: ver,
+		Date:    time.Now(),
+	})
+	if err != nil {
+		return err
+	}
+	log.Debugf(ctx, "Writing installed version info to: %s", versionFile)
+	return os.WriteFile(versionFile, raw, 0o600)
+}
+
+func (p *Project) checkUpdates(cmd *cobra.Command, _ []string) error {
+	ctx := cmd.Context()
+	r := releases.NewReleaseCache("databrickslabs", p.Name, p.CacheDir(ctx))
+	versions, err := r.Load(ctx)
+	if err != nil {
+		return err
+	}
+	installed, err := p.InstalledVersion(ctx)
+	if err != nil {
+		return err
+	}
+	latest := versions[0]
+	if installed.Version == latest.Version {
+		return nil
+	}
+	ago := time.Since(latest.PublishedAt)
+	// TODO: make bold
+	cmd.PrintErrf("[UPGRADE ADVISED] You're using %s (%s), which has got newer version: %s %s ago. "+
+		"Please run `databricks labs upgrade %s` to upgrade.",
+		p.Name, installed.Version, latest.Version, ago, p.Name)
+	return nil
+}
+
+func (p *Project) profileOverride(cmd *cobra.Command) string {
+	profileFlag := cmd.Flag("profile")
+	if profileFlag == nil {
+		return ""
+	}
+	return profileFlag.Value.String()
+}
+
+type version struct {
+	Version string    `json:"version"`
+	Date    time.Time `json:"date"`
+}
+
+type authConfig struct {
+	WorkspaceProfile string `json:"workspace_profile,omitempty"`
+	AccountProfile   string `json:"account_profile,omitempty"`
+}
+
+func (ac *authConfig) ConfigForWorkspace(ctx context.Context) (*config.Config, error) {
+	cfg := &config.Config{
+		ConfigFile: filepath.Join(env.UserHomeDir(ctx), ".databrickscfg"),
+		Profile:    ac.WorkspaceProfile,
+	}
+	return cfg, cfg.EnsureResolved()
+}
+
+func (ac *authConfig) ConfigForAccount(ctx context.Context) (*config.Config, error) {
+	cfg := &config.Config{
+		ConfigFile: filepath.Join(env.UserHomeDir(ctx), ".databrickscfg"),
+		Profile:    ac.AccountProfile,
+	}
+	err := cfg.EnsureResolved()
+	if err != nil {
+		return nil, err
+	}
+	if !cfg.IsAccountClient() {
+		return nil, databricks.ErrNotAccountClient
+	}
```

Comment:
We also check this at prompt time. Is it a dup?

Otherwise, also replicate this check for workspace client (and check for `cfg.IsWorkspaceClient()`).

Reply:
simplified a bit, now no longer a dup